.github/workflows/generate-monthly-updates-template.yml

```diff
@@ -0,0 +1,56 @@
+name: Generate Monthly Updates Blog Post Template
+on:
+  schedule:
+    - cron: '0 0 1 * *'  # Run on the first day of every month
+  workflow_dispatch:
+
+jobs:
+  generate-post:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install script dependencies
+        run: |
+          pip install requests
+
+      - name: Generate blog post and raise PR
+        env:
+          GITHUB_USER: zapbot
+          GITHUB_TOKEN: ${{ secrets.ZAPBOT_TOKEN }}
+          BRANCH_NAME: monthly-updates-blog-post
+        run: |
+          current_date=$(date +%Y-%m-%d)
+          day_of_month=$(date +%d)
+          if [ "$day_of_month" -ge 10 ]; then
+            month_year=$(date +%B-%Y)  # Current month and year
+          else
+            month_year=$(date --date='-1 month' +%B-%Y)  # Previous month and year
+          fi
+          month_year=$(echo "$month_year" | tr '[:upper:]' '[:lower:]')
+          DIR="site/content/blog/${current_date}-zap-updates-${month_year}"
+          mkdir -p $DIR
+          pushd $DIR
+          python tools/monthly-updates-blog-post/generate-blog-post.py > index.md
+          popd
```

Comment:
I'm not necessarily against this but why not cd?

Reply:
No particular reason, but I just realized we don't really need to change directories at all. Will update this.